demos/http/http_demo_s3_download_async/http_demo_s3_download_async.c

```diff
@@ -0,0 +1,751 @@
+                }
+            }
+            else
+            {
+                LogError( ( "Failed to send HTTP request: Error=%s.",
+                            HTTPClient_strerror( httpStatus ) ) );
+            }
+        }
+    }
+    else if( forkPID == -1 )
+    {
+        LogError( ( "Error forking." ) );
+    }
+
+    return forkPID;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Entry point of demo.
+ *
+ * This example resolves a domain, establishes a TCP connection, validates the
+ * server's certificate using the root CA certificate defined in the config header,
+ * then finally performs a TLS handshake with the HTTP server so that all communication
+ * is encrypted. After which, HTTP Client library API is used to download the
+ * S3 file by sending multiple GET requests, filling up the response buffer
+ * each time until all parts are downloaded. If any request fails, an error
+ * code is returned.
+ *
+ * @note This example is single-threaded and uses statically allocated memory.
+ *
+ */
+int main( int argc,
+          char ** argv )
+{
+    /* Return value of main. */
+    int32_t returnStatus = EXIT_SUCCESS;
+    /* HTTPS Client library return status. */
+    HTTPStatus_t httpStatus = HTTP_SUCCESS;
+
+    /* The location of the path within string S3_PRESIGNED_URL. */
+    const char * pPath = NULL;
+    /* The length of the path within string S3_PRESIGNED_URL. */
+    size_t pathLen = 0;
+
+    /* The transport layer interface used by the HTTP Client library. */
+    TransportInterface_t transportInterface = { 0 };
+    /* The network context for the transport layer interface. */
+    NetworkContext_t networkContext = { 0 };
+
+    /* Queue for http requests to be handled by http thread */
+    mqd_t requestQueue = -1;
+    /* Queue for http responses retrieved by http thread */
+    mqd_t responseQueue = -1;
+    /* Settings for constructing queues */
+    struct mq_attr queueSettings;
+
+    /* Number of requests made */
+    size_t requestCount = 0;
+
+    /* PID of HTTP thread. */
+    pid_t httpThread = -1;
+
+    ( void ) argc;
+    ( void ) argv;
+
+    for( ; ; )
+    {
+        LogInfo( ( "HTTPS Client Asynchronous S3 download demo using pre-signed URL:\n%s", S3_PRESIGNED_URL ) );
+
+        /**************************** Parse Signed URL. ******************************/
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            /* Retrieve the path location from S3_PRESIGNED_URL. This function returns the length of the path
+             * without the query into pathLen. */
+            httpStatus = getUrlPath( S3_PRESIGNED_URL,
+                                     strlen( S3_PRESIGNED_URL ),
+                                     &pPath,
+                                     &pathLen );
+
+            if( httpStatus != HTTP_SUCCESS )
+            {
+                LogError( ( "An error occurred in getUrlPath() on URL %s. Error code: %d",
```

Comment:
I think https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1318 is changing this to log inside the URL parsing functions so we can delete these logs.